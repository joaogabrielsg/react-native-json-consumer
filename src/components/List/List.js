import React from 'react';
import PropTypes from 'prop-types';
import { View, FlatList, StyleSheet } from 'react-native';

const List = ({ listItems }) => (
  <View style={styles.container}>
    <FlatList
      data={listItems}
      renderItem={item => {
        return <View style={styles.list}>{item.item}</View>;
      }}
    />
  </View>
);

const styles = StyleSheet.create({
  container: {
    flex: 1,
    width: '100%'
  },
  list: {
    marginBottom: 10,
    justifyContent: 'center',
    alignItems: 'center'
  }
});

List.propTypes = {
  listItems: PropTypes.array.isRequired
};

export default List;
