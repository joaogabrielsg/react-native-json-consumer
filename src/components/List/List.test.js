import 'react-native';
import React from 'react';
import List from './List';

import renderer from 'react-test-renderer';

describe('List', () => {
  it('sholud render correctly', () => {
    const list = renderer.create(<List listItems={[]} />).toJSON();
    expect(list).toMatchSnapshot();
  });
});
