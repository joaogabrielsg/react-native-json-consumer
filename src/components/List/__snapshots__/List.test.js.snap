// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List sholud render correctly 1`] = `
<View
  style={
    Object {
      "flex": 1,
      "width": "100%",
    }
  }
>
  <RCTScrollView
    data={Array []}
    disableVirtualization={false}
    getItem={[Function]}
    getItemCount={[Function]}
    horizontal={false}
    initialNumToRender={10}
    keyExtractor={[Function]}
    maxToRenderPerBatch={10}
    numColumns={1}
    onContentSizeChange={[Function]}
    onEndReachedThreshold={2}
    onLayout={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    renderItem={[Function]}
    scrollEventThrottle={50}
    stickyHeaderIndices={Array []}
    updateCellsBatchingPeriod={50}
    viewabilityConfigCallbackPairs={Array []}
    windowSize={21}
  >
    <View />
  </RCTScrollView>
</View>
`;
